### PR TITLE
Implement the composite search bar for the mobile search header

### LINKED ISSUE #1876: Component: VHeaderMobile
## Description

The new mobile search header is a composite component. In resting state, it shows the Openverse logo link, the search term, and the `VContentSettingsButton`. When the search input is activated, it shows a back button in the start (left) of the input, and a cross button for clearing the search term. 
When the search input is activated, the Recent search terms modal should be opened. 
The PR for this issue does not have to implement the Recent searches modal and the Content settings modal

### Slots

| Name | Slot props | Description |
|-|-|-|
| start | - | The items at the start of the header: VLogoButton when resting, and VBackButton when active
| default | - | The items at the end of the header: the VContentSettingsButton when resting and VClearButton when active

### State

The functionality for handling the search submission and search text updates should be extracted to a separate composable, and should be shared with the `VHeaderDesktop` (TODO: create the issue for the composable)

## Code samples

```
<header>
  <form>
    <slot name="start">
      <VLogoButton v-if="!isActive"/>
      <VBackButton v-else />
    </slot>
    <input id="search-input" />
    <slot>
      <VClearButton v-if="isActive" />
      <template v-else>
        <VSearchStatus class="hidden md:flex" /> // The number of results
        <VSearchContentSettingsButton />
      </template>
    </slot>
  </form>
</header>
```

## References

- **Figma:**
- Resting: https://www.figma.com/file/KxkZmBNk218V2Fke1GeYCu/New-header-component?node-id=552%3A79971
- Active: https://www.figma.com/file/KxkZmBNk218V2Fke1GeYCu/New-header-component?node-id=552%3A79090

## Implementation

- [x] 🙋 I would be interested in implementing this component.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1876 by @obulat

## Description

This PR implements the new VHeaderMobile component that is used on mobile widths for the search result and single result pages.
It does not use the `VSearchBar` or `VInputField` components, copying some code from them instead. Re-using the existing components is difficult because of the different border treatments (in the current component, everything is inside an outer border, and in the existing VSearchBar, some components are outside of the border), and nesting several components and several levels of template `ref`s is difficult to follow. We could extract the common functionality in the future, if that makes the code simpler.

## Testing Instructions

Go to 'http://localhost:8443/search/?q=cat&ff_new_header=on' on a narrow screen. The search bar should behave correctly. You should be able to update the search term, clear it, open (a stub of) the `VContentSettingsModal`.

You can also check out the VHeaderMobile story: https://wordpress.github.io/openverse-frontend/_preview/1875/?path=/story/components-vheader-vheadermobile-vheadermobile--default-story

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VHeader/VHeaderMobile/VBackButton.vue [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<template>
+  <VButton
+    size="disabled"
+    class="icon-button flex h-12 w-12 flex-shrink-0 rotate-90 items-center justify-center bg-tx active:shadow-ring"
+    variant="plain--avoid"
+    aria-label="Go back"
```
Comment (zackkrida) [inline src/components/VHeader/VHeaderMobile/VBackButton.vue:6]:
```suggestion
    aria-label="Go back"
```

Should this be added to `en.json`?

File: src/components/VHeader/VHeaderMobile/VContentSettingsModal.vue [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          ref="searchTypesRef"
+          active-item="image"
+          size="small"
+          :use-links="true"
+        />
+      </nav>
+    </template>
+  </VModal>
+</template>
+<script lang="ts">
```
Comment (zackkrida) [inline src/components/VHeader/VHeaderMobile/VContentSettingsModal.vue:31]:
```suggestion
</template>

<script lang="ts">
```

File: src/components/VHeader/VSearchBar/VClearButton.vue [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<template>
+  <VIconButton
+    :button-props="{ variant: 'plain' }"
+    :icon-props="{ iconPath: closeIcon }"
+    size="smaller"
+    class="border-tx text-dark-charcoal-70"
+    :aria-label="$t('browse-page.search-form.clear')"
```
Comment (zackkrida) [inline src/components/VHeader/VSearchBar/VClearButton.vue:7]:
```vue
    :aria-label="$t('browse-page.search-form.clear')"
```

Curiously this doesn't appear to be working:

CleanShot 2022-10-04 at 08 39 00@2x

File: tsconfig.json [hunk 4]
```diff
@@ -78,7 +78,7 @@
     "src/components/VErrorSection/VErrorImage.vue",
     "src/components/VFilters/VLicenseExplanation.vue",
     "src/components/VHeader/VFilterButton.vue",
-    "src/components/VHeader/VSearchBar/**.vue",
+    //    "src/components/VHeader/VSearchBar/**.vue",
```
Comment (zackkrida) [inline tsconfig.json:81]:
```suggestion
    //    "src/components/VHeader/VSearchBar/**.vue",
```

Comments in json 😢

**Review by zackkrida — CHANGES_REQUESTED** [review]
Very close to lgtm. Two things I noticed (one we can wait for @panchovm) on:

- The Openverse logo and the back button have different alignments so when you click on the input the text moves to the right slightly.
- The 'clear' button appears and is active even if there is no text in the input.

File: src/components/VHeader/VHeaderMobile/VBackButton.vue (hunk 1 above)
Comment (obulat) [reply src/components/VHeader/VHeaderMobile/VBackButton.vue:6]: Definitely! Thank you for catching.

File: tsconfig.json (hunk 4 above)
Comment (obulat) [reply tsconfig.json:81]: I didn't even realize that this is a `json` file!

File: src/components/VHeader/VSearchBar/VClearButton.vue (hunk 3 above)
Comment (obulat) [reply src/components/VHeader/VSearchBar/VClearButton.vue:7]: I forgot to add the string to `en.json` in this branch 😬

**Review by zackkrida — CHANGES_REQUESTED** [review]
- The changes to the logo/back button jumping look nice 👍 the logo is larger than advised by the mockups but when I test making it smaller locally it looks quite odd. We can review this with @panchovm later in the month.
- The 'clear' button still appears when there is no search term present. I believe this is fixed in #1880 so as long as that continues I'm okay with this
- A small thing: the back button has no interaction or focus styles. Can this be updated with the universal styles from the button audit you completed too?

**Comment by obulat** [comment]
Changes since the last review:
- the active search bar border was replaced: now using the `ring-pink` instead of `border-pink` (so, it's now external instead of internal)
- the buttons were refactored to have a full tappable width and height of 48x48px to comply with a11y requirements. The inner colored area is still 32x32px to match the mockups. 
- Hover and focus styles added to the buttons.
- 'Clear' button is removed when there is no search term.

**Review by dhruvkb — APPROVED** [review]

**Review by zackkrida — APPROVED** [review]
LGTM. I think I see some bugs that aren't related to the contents of this PR, which I will make separate issues for.

**Pull request merged**